src/liboslexec/llvm_gen.cpp

```diff
@@ -386,6 +388,20 @@ LLVMGEN (llvm_gen_printf)
         }
     }
 
+    if (rop.use_optix() && arg > (format_arg + 2)) {
+        std::cerr << "WARNING: "
```

Comment:
Note to self: I never thought to do it before, but I bet I could "constant fold" printfs so that quite often they would collapse to 0 or 1 true runtime arguments.